src/redis-check-aof.c

```diff
@@ -51,45 +67,138 @@ int consumeNewline(char *buf) {
+        if (fgets(buf, AOF_MANIFEST_MAX_LINE+1, fp) == NULL) {
+            if (feof(fp)) {
+                if (linenum == 0) {
+                    err = "Found an empty AOF manifest";
+                    goto loaderr;
                 } else {
-                    printf("Successfully truncated AOF to timestamp %ld\n",
-                            to_timestamp);
-                    fclose(fp);
-                    exit(0);
+                    break;
                 }
+            } else {
+                err = "Read AOF manifest failed";
+                goto loaderr;
             }
-            continue;
-        } else {
-            if (fseek(fp, -(ftello(fp)-epos), SEEK_CUR) == -1) {
-                ERROR("Fseek error: %s", strerror(errno));
-                return 0;
+        }
+
+        linenum++;
+
+        /* Skip comments lines */
+        if (buf[0] == '#') continue;
+
+        if (strchr(buf, '\n') == NULL) {
+            err = "The AOF manifest file contains too long line";
+            goto loaderr;
+        }
+
+        line = sdstrim(sdsnew(buf), " \t\r\n");
+        if (!sdslen(line)) {
+            err = "The AOF manifest file is invalid format";
+            goto loaderr;
+        }
+
+        argv = sdssplitargs(line, &argc);
+        /* 'argc < 6' was done for forward compatibility. */
+        if (argv == NULL || argc < 6 || (argc % 2)) {
+            err = "The AOF manifest file is invalid format";
+            goto loaderr;
+        }
+
+        ai = zcalloc(sizeof(aofInfo));
+        for (int i = 0; i < argc; i += 2) {
+            if (!strcasecmp(argv[i], AOF_MANIFEST_KEY_FILE_NAME)) {
+                ai->file_name = sdsnew(argv[i+1]);
+                if (!pathIsBaseName(ai->file_name)) {
+                    err = "File can't be a path, just a filename";
+                    goto loaderr;
+                }
+            } else if (!strcasecmp(argv[i], AOF_MANIFEST_KEY_FILE_SEQ)) {
+                ai->file_seq = atoll(argv[i+1]);
+            } else if (!strcasecmp(argv[i], AOF_MANIFEST_KEY_FILE_TYPE)) {
+                ai->file_type = (argv[i+1])[0];
             }
-            return 1;
+            /* else if (!strcasecmp(argv[i], AOF_MANIFEST_KEY_OTHER)) {} */
+        }
+
+        if (!ai->file_name || !ai->file_seq || !ai->file_type) {
+            err = "The AOF manifest file is invalid format";
+            goto loaderr;
         }
+
+        sdsfreesplitres(argv, argc);
+        argv = NULL;
+
+        if (ai->file_type == AOF_FILE_TYPE_BASE) {
+            if (am->base_aof_info) {
+                err = "Found duplicate base file information";
+                goto loaderr;
+            }
+            am->base_aof_info = ai;
+            am->curr_base_file_seq = ai->file_seq;
+        } else if (ai->file_type == AOF_FILE_TYPE_HIST) {
+            listAddNodeTail(am->history_aof_list, ai);
+        } else if (ai->file_type == AOF_FILE_TYPE_INCR) {
+            if (ai->file_seq <= maxseq) {
```

Comment:
Can't we assume the sequence always grows by 1?
```suggestion
            if (ai->file_seq != maxseq+1) {
```

Reply:
In theory you are right, but I don't want to add such a strong limit here, because we just load the corresponding files in order according to the instructions of the manifest, and the increasing order is guaranteed to ensure that we can get a maximum value. 
Too restrictive means less scalability later.